src/core/link-to-dfn.js

```diff
@@ -56,23 +56,27 @@ export async function run(conf) {
   /** @type {HTMLAnchorElement[]} */
   const badLinks = [];
 
-  const localLinkSelector =
-    "a[data-cite=''], a:not([href]):not([data-cite]):not(.logo):not(.externalDFN)";
-  document.querySelectorAll(localLinkSelector).forEach((
-    /** @type {HTMLAnchorElement} */ anchor
-  ) => {
+  /** @type {NodeListOf<HTMLAnchorElement>} */
+  const localAnchors = document.querySelectorAll(
+    "a[data-cite=''], a:not([href]):not([data-cite]):not(.logo):not(.externalDFN)"
+  );
+  for (const anchor of localAnchors) {
     const linkTargets = getLinkTargets(anchor);
-    const foundDfn = linkTargets.some(target => {
-      return findLinkTarget(target, anchor, titleToDfns);
-    });
-    if (!foundDfn && linkTargets.length !== 0) {
+    const linkTarget = linkTargets.find(
+      target =>
+        titleToDfns.has(target.title) &&
+        titleToDfns.get(target.title).has(target.for)
+    );
+    if (linkTarget) {
+      useLinkTarget(linkTarget, anchor, titleToDfns);
+    } else if (linkTargets.length !== 0) {
```

Comment:
How about:

```suggestion
    } else if (!linkTargets.length) {
```

I think this pattern is used frequent enough.

Reply:
Just found out `else {` works also - unless we're missing some tests